sequence_processing_pipeline/tests/test_Pipeline.py

```diff
@@ -1,72 +1,148 @@
 
-        configuration_file = 'sequence_processing_pipeline/configuration.json'
-
-        # shutil.rmtree(
-        # 'sequence_processing_pipeline/tests/data/invalid_output_directory')
-        # shutil.rmtree(('sequence_processing_pipeline/tests'
-        #               '/data/invalid_output_directory_two'),
-        #              ignore_errors=False)
-
-        # test invalid input_directory
-        self.assertRaises(PipelineError,
-                          Pipeline,
-                          configuration_file,
-                          'not-valid-sample-sequence-directory',
-                          'tests/data/output_directory')
-
-        # test non-existant output_directory
-        # pipeline = Pipeline(input_directory,
-        #                    inv_output_directory,
-        #                    final_output_directory)
-        # self.assertTrue(exists(inv_output_directory))
-
-        # test non-existant final_output_directory
-        # pipeline = Pipeline(input_directory,
-        #                    output_directory,
-        #                    inv_final_output_directory)
-        # self.assertTrue(exists(inv_final_output_directory))
-
-        # nprocs exceeds 16 nproc limit
-        # self.assertRaises(PipelineError, Pipeline, input_directory,
-        #                  output_directory, final_output_directory, nprocs=18)
-
-        # nprocs must be greater than 0.
-        # self.assertRaises(PipelineError, Pipeline, input_directory,
-        #                  output_directory, final_output_directory, nprocs=0)
-
-        # window of time must be legitimate.
-        # (older_than can't be less than or equal to younger_than.)
-        # self.assertRaises(PipelineError, Pipeline, input_directory,
-        #                   output_directory, final_output_directory,
-        #                   younger_than=24, older_than=24)
-
-        # using all default parameters should not raise an Error.
-        test_path = partial(join, dirname(abspath(__file__)))
-        input_directory = test_path('data/sample-sequence-directory')
-
-        msg = None
-        try:
-            Pipeline(configuration_file, input_directory, run_id='run-id')
-        except PipelineError as e:
-            msg = str(e)
-
-        self.assertEqual(msg, None)
+        self.good_config_file = ('sequence_processing_pipeline/'
+                                 'configuration.json')
+        self.bad_config_file = ('sequence_processing_pipeline/tests/data/'
+                                'bad_configuration.json')
+        self.invalid_config_file = 'does/not/exist/configuration.json'
+        self.good_run_id = 'sample-sequence-directory'
+        self.invalid_run_id = 'not-sample-sequence-directory'
+        self.data_directory = 'sequence_processing_pipeline/tests/data'
+        self.maxDiff = None
+
+    def test_creation(self):
+        # Pipeline should assert due to config_file
+        with self.assertRaises(PipelineError):
```

Comment:
For these tests, I think we should add a tests for the error message to make sure they make sense; basically:
```python
with self.assertRaises(XXXXX) as err:
     code
self.assertEqual(str(err.exception), "Error message")
```

Reply:
done